compute/createInstanceWithLocalSSD.js

```diff
@@ -0,0 +1,183 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function main(projectId, zone, instanceName) {
+  // [START compute_instances_create_with_local_ssd]
+  // TODO(developer): Uncomment and replace these variables before running the sample.
+  // const projectId = 'YOUR_PROJECT_ID';
+  // const zone = 'europe-central2-b';
+  // const instanceName = 'YOUR_INSTANCE_NAME';
+
+  const compute = require('@google-cloud/compute');
+
+  // Create a new VM instance with Debian 10 operating system and SSD local disk.
+  async function createInstanceWithLocalSSD() {
+    const diskSizeGb = 10;
+    const boot = true;
+    const autoDelete = true;
+    const diskType = `zones/${zone}/diskTypes/pd-standard}`;
+    const disks = [];
+
+    // Get the latest debian image.
+    // Retrieve the newest image that is part of a given family in a project.
+    // Args:
+    //    project: project ID or project number of the Cloud project you want to get image from.
+    //    family: name of the image family you want to get image from.
+    const imagesClient = new compute.ImagesClient();
+    const [newestDebian] = await imagesClient.getFromFamily({
+      project: 'debian-cloud',
+      family: 'debian-11',
+    });
+
+    // Create the disks to be included in the instance.
+    disks.push(
+      await createDiskFromImage(
+        diskType,
+        diskSizeGb,
+        boot,
+        newestDebian.selfLink,
+        autoDelete
+      )
+    );
+    disks.push(await createLocalSsdDisk(zone));
+
+    // Create the instance.
+    const [instance] = await createInstance(
+      projectId,
+      zone,
+      instanceName,
+      disks
+    );
+
+    if (instance !== null) {
+      console.log(`Instance created with local SSD: ${instance.name}`);
+    }
+  }
+
+  // Create an AttachedDisk object to be used in VM instance creation. Uses an image as the
+  // source for the new disk.
+  //
+  // Args:
+  //    diskType: the type of disk you want to create. This value uses the following format:
+  //        "zones/{zone}/diskTypes/(pd-standard|pd-ssd|pd-balanced|pd-extreme)".
+  //        For example: "zones/us-west3-b/diskTypes/pd-ssd"
+  //
+  //    diskSizeGb: size of the new disk in gigabytes.
+  //
+  //    boot: boolean flag indicating whether this disk should be used as a
+  //    boot disk of an instance.
+  //
+  //    sourceImage: source image to use when creating this disk.
+  //    You must have read access to this disk. This can be one of the publicly available images
+  //    or an image from one of your projects.
+  //    This value uses the following format: "projects/{project_name}/global/images/{image_name}"
+  //
+  //    autoDelete: boolean flag indicating whether this disk should be deleted
+  //    with the VM that uses it.
+  async function createDiskFromImage(
```

Comment:
Why would this and other functions be `async`? They don't have anything asynchronous in them, so they should be synchronous and used without `await`.

Reply:
Thanks for the comment.

I've removed async from a couple of struct type functions. The other function to create an instance does require async.